lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -265,6 +308,7 @@ def _on_peer_relation_changed(self, event: RelationChangedEvent):
         ):
             # we defer because we want the temporary status to be updated
             event.defer()
+            self.on.update_status.emit()
```

Comment:
This was to ensure the deferred event will be replayed (as opposed to waiting for the potential next event in the queue) - this is no longer applicable, as we adopted a plain dummy [`DeferEventTrigger`](https://github.com/canonical/opensearch-operator/blob/main/lib/charms/opensearch/v0/opensearch_base_charm.py#L113C39-L113C56)

Reply:
If you replay an event immediately, it will have the same relation-data context during execution as the event that `emitted` it. 
What are you expecting to have changed? Is it some API call?

Reply:
Yes, it's `self.health.apply()` - which essentially fetches (through the OpenSearch REST API) the health of the cluster